### PR TITLE
Project Cluedo - Jamie Cook

### PR DESCRIPTION
**Pull request by the-j-curl**
I completed the blue requirements. I also completed all of the red level goals. Which included:
Adding favouriteWeapon property, and making this a random selection. 
I also added a loading animation.

I had trouble trying to get the shuffleFavouriteWeapon function working. And I also spent a lot of time getting the animation to work. I found that after I added the animation my RESET button stopped working. I have left this in my product to hopefully get some feedback on. 
Another strange issue I came across was when declaring my revealMystery function it only worked when I created it using the function keyword, and did not execute if I declared it as: const revealMystery = () => { 
(see line 334 in script.js). I posted this as a question on SO FYI.

I also feel a little confused with the JavaScript once the code starts looking more complex. 
I would of liked to have had more time so I could have created a new array as part of the black level goals.

### CODE REVIEW

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: Missing semi-colon

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: Missing semi-colon

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: Missing semi-colon

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: You don't need this line since you are using "onclick" in the HTML-file. Either you keep this in the JS-file and remove the "onclick" or vice versa.

Please update you PR and I will later on approve it!

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: You don't need this line since you are using "onclick" in the HTML-file. Either you keep this in the JS-file and remove the "onclick" or vice versa.

Please update you PR and I will later on approve it!

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: You don't need this line since you are using "onclick" in the HTML-file. Either you keep this in the JS-file and remove the "onclick" or vice versa.

If you want to keep it in the JS-file you need to change `roomcard` to `roomCard` to get it to work. If you goto your site and look at the Console you will see that it complaining about this row.

Please update you PR and I will later on approve it!

File: code/clue.js
Comment (ylvan75) [inline code/clue.js:280]: Use arrow notation for functions `setTimeout(() => {`

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: Use arrow notation for functions `setTimeout(() => {`

File: code/clue.js
Comment (ylvan75) [inline code/clue.js:316]: Use arrow notation for functions `setTimeout(() => {`

File: code/clue.js
Comment (ylvan75) [inline code/clue.js]: A tip if you would have wanted a message to be presented for the user if you try to click "Reveal the crime" before all deck cards where picked.

```suggestion
  if (
    mystery.killer === '' ||
    mystery.weapon === '' ||
    mystery.room === ''
  ) {
    document.getElementById("mystery").innerHTML =
      "Click all cards above to reveal the crime!";
  } else {
    document.getElementById(
      "mystery"
    ).innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} room with a ${mystery.weapon.name}!`;
```

File: code/clue.js
Comment (ylvan75) [inline code/clue.js:329]: As you have mentioned you want to use arrow notation for functions but it didn't work for you. I don't see what you need to do to get it to work. Wondering if it can have something to do that you do onclick on a html-tag ` <button>`

File: code/clue.js
Comment (ylvan75) [inline code/clue.js:337]: As you have mentioned you want to use arrow notation for functions but it didn't work for you. I don't see what you need to do to get it to work. Wondering if it can have something to do that you do onclick on a html-tag ` <button>`

File: code/index.html
Comment (ylvan75) [inline code/index.html:25]: I'm just curious, when is this one used?

File: code/index.html
Comment (ylvan75) [inline code/index.html:34]: I'm just curious, when is this one used?

File: code/index.html
Comment (ylvan75) [inline code/index.html:42]: I'm just curious, when is this one used?

File: code/style.css
Comment (ylvan75) [inline code/style.css]: Good that you group them like this, since they use same styling

File: code/style.css
Comment (ylvan75) [inline code/style.css]: Good that you group them like this, since they use same styling

File: code/style.css
Comment (ylvan75) [inline code/style.css:139]: Nice effect!

File: code/style.css
Comment (ylvan75) [inline code/style.css:144]: Nice effect!

**Review by ylvan75 — CHANGES_REQUESTED** [review]
Great job Jamie :-)

I really liked your Pull Request description (got me inspired to make mine better in the future). It gave me a good overview of what level you aimed for and what you have been struggling with during the project. Nice that you have highlighted things as the RESET button and revealMystery function. Made it easier for me when I started to test your website and later on looking into the code.

It was also good that you in the README said that you focused om the JavaScript and didn't put in time for the responsiveness part. Good decision!

I like your background image and that you have added images for weapon and room, the deck cards looks more balanced then.

I notice when I tested your site that I didn't get any message with instructions what to do if I tried to "Reveal the crime" before all deck cards where picked. In your JS-file at the function revealMystery I gave you a tip on how you could have done it, if that would have been something you wanted to do.

Don't forget to be consistent if you want to use semi-colon. I found a couple of places in the JS-file that the semi-colon is missing, some examples line 257, 258, 279 and 324.

All three HTML, CSS and JS-files are really clean and made it easy to review.

Before being approved decide if you wanna use `onclick` in the HTML-file or `addEventListener` in JS-file, see my comment in the JS-file on line 296, 313 and 329.

File: code/index.html
Comment (the-j-curl) [reply code/index.html:42]: This div is my card loader animation

File: code/clue.js
Comment (the-j-curl) [reply code/clue.js]: Thanks - well spotted. I have removed the eventListener

**Comment by the-j-curl** [comment]
Thanks for the feedback Ylva. Some useful comments.
I have added the missing semi-colons to be consistent and removed the event Listener which wasn't being used.

**Review by ylvan75 — APPROVED** [review]

**Comment by ylvan75** [comment]
Thumbs up from me!

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js]: Good idea to store this info in variables! Nonetheless, one thing could be improved :
There is a strict convention about boolean variables names. They should always start with `is` as a prefix: `isKillerPicker`, `isWeaponPicked`, `isRoomPicked`

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js]: You don't overwrite those values anywhere in app, they should be declared with `const`

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js]: It's even more semantic approach to initialise mystery object with `null` values like this :

```
const mystery = {
  killer: null,
  room: null,
  weapon: null
}
```

Moreover, you can use `const` keyword - we will be changing values of properties, that's true, but JavaScript will not consider this as overwriting. What counts for JavaScript is that object will stay object

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js:271]: wrong indentation

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js]: very nice Jamie :)

File: code/clue.js
Comment (EmOlsz) [inline code/clue.js]: Or we can simplify that statement :

```
if (killerIsPicked && weaponIsPicked && roomIsPicked) { ... }
```

File: code/style.css
Comment (EmOlsz) [inline code/style.css]: It's far more safe approach to wrap path in parentheses. Without them, some browsers might not find the path
`url("./assets/image-main-dark5.jpg");`

File: code/style.css
Comment (EmOlsz) [inline code/style.css:113]: I'm happy you use classes only in your CSS. In general, your CSS code is very clean, great job :)

File: code/style.css
Comment (EmOlsz) [inline code/style.css]: These properties never change, no need to add them in `@keyframes`. The best approach would be to add them to `.loader` class

**Review by EmOlsz — APPROVED** [review]
I'm very happy about quality of your code Jamie! It's well thought, consequent and aesthetic (however indentation could be improved). Especially CSS part caught my attention :)
Take a look on my comments to improve even more, nonetheless project is approved!

**Comment by the-j-curl** [comment]
@EmOlsz - Thanks for all the feedback Maks. I have updated my code as per your advice and will carry this info over to future projects.

**Pull request closed without merging**
